app/src/test/java/io/intrepid/contest/screens/conteststatus/ContestStatusPresenterTest.java

```diff
@@ -64,34 +76,46 @@ public void onViewCreatedShouldShowWaitingSubmissionsFragmentWhenStatusIsWaiting
         presenter.onViewCreated();
         testConfiguration.triggerRxSchedulers();
 
-        verify(mockView).showWaitingSubmissionsFragment(response.numSubmissionsMissing);
+        verify(mockView).showWaitingSubmissionsFragment(response.getNumSubmissionsMissing());
         verify(mockView, never()).showResultsAvailableFragment();
+        verify(mockView, never()).showContestOverviewPage();
     }
 
     @Test
-    public void onViewCreatedShouldShowResultsAvailableFragmentWhenStatusIsNotWaitingForSubmissions() {
-        ContestStatusResponse response = new ContestStatusResponse();
-        response.waitingForSubmissions = false;
-        response.numSubmissionsMissing = 0;
-        when(mockRestApi.getContestStatus(any())).thenReturn(Observable.just(response));
+    public void onViewCreatedShouldShowContestOverviewWhenWaitingForScoresAndParticipantIsJudge() {
+        when(mockPersistentSettings.getCurrentParticipationType()).thenReturn(ParticipationType.JUDGE);
+        getContestStatusResponseWaitingForScores();
+
+        presenter.onViewCreated();
+        testConfiguration.triggerRxSchedulers();
+
+        verify(mockView).showContestOverviewPage();
+        verify(mockView, never()).showWaitingSubmissionsFragment(any(int.class));
```

Comment:
Will do! I've been using `any(__.class)` for all primitives, at some point I can refactor!